sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java

```diff
@@ -24,41 +24,181 @@
+   * UnsupportedOperationException if the configuration for applications or groups are not set.
+   *
+   * @param appName
+   * @param group
+   * @return the group as it has been passed
+   */
+  Group createGroup(String appName, Group group);
 
-  Organization updateOrganization(Organization updatedOrganization);
+  /**
+   * Replace the group of same name in the application appName by updatedGroup. Users are unchanged.
+   * Throws UnsupportedOperationException if the configuration for applications or groups are not
+   * set.
+   *
+   * @param appName
+   * @param updatedGroup
+   * @return the group as it has been passed.
+   */
+  Group updateGroup(String appName, Group updatedGroup);
 
-  void deleteUserFromGroup(String appName, String groupName, String userId);
+  /**
+   * Delete the group groupName of the application appName in the store. Users remain unchanged.
+   * Throws UnsupportedOperationException if the configuration for applications or groups are not
+   * set.
+   *
+   * @param appName
+   * @param groupName
+   */
+  void deleteGroup(String appName, String groupName);
 
+  /**
+   * Add a user to the group groupName in the application appName. The user might not exist on the
+   * realm. Throws UnsupportedOperationException if the configuration for applications or groups is
+   * not set.
+   *
+   * @param appName
+   * @param groupName
+   * @param userId
+   */
   void addUserToGroup(String appName, String groupName, String userId);
 
-  void reinitPassword(
-      User user, String generatedPassword, PasswordChangeRequest pcr, List<SendMode> sendModes);
+  /**
+   * Delete the user userId from the group groupName in the application appName. Throws
+   * UnsupportedOperationException if the configuration for applications or groups is not set.
+   *
+   * @param appName
+   * @param groupName
+   * @param userId
+   */
+  void deleteUserFromGroup(String appName, String groupName, String userId);
 
+  /**
+   * Set the password of user to initPassword in a raw format. If password already exist, changes
+   * it.
+   *
+   * @param user
+   * @param initPassword, password to set to the user, cannot be null
+   * @param pcr not used
+   * @param sendModes not used
+   */
   void initPassword(
       User user, String initPassword, PasswordChangeRequest pcr, List<SendMode> sendModes);
 
+  /**
+   * Set the password of user to generatedPassword in a raw format. Same behaviour than
+   * initPassword.
+   *
+   * @param user
+   * @param generatedPassword password to set to the user, cannot be null
+   * @param pcr not used
+   * @param sendModes not used
+   */
+  void reinitPassword(
+      User user, String generatedPassword, PasswordChangeRequest pcr, List<SendMode> sendModes);
+
+  /**
+   * Change the user password from oldPassword to newPassword. OldPassord and newPassword are
+   * considered as raw even though oldPassword is an hash. Throws InvalidPasswordException if
```

Comment:
> even though oldPassword is an hash 

Do you mean if it is stored as an hash ? I don't think it is important here, what we want is the correct oldPassword not the hash, the storage implementation is not important (depends on the differents store providers)

Reply:
I meant that you if you have a password stored as {SHA}c3q3RSeNwMY7E09Ve9oBHw+MVXg= the password to pass to validate the credential should be {SHA}c3q3RSeNwMY7E09Ve9oBHw+MVXg=. But yes it is irrelevant here.

Reply:
this is not normal, even with a password stored as a hash (as it should be) I would expect the oldPassword to pass to be the real value. If not I'm guessing we don't validate it like we should (with ldap we must validate password by a bind operation, attribute value comparison is very very wrong).